app/src/test/java/org/simple/clinic/registration/phone/RegistrationPhoneScreenControllerTest.kt

```diff
@@ -120,6 +126,70 @@ class RegistrationPhoneScreenControllerTest {
   @Test
   fun `when input text is changed then any visible errors should be removed`() {
     uiEvents.onNext(RegistrationPhoneNumberTextChanged(""))
-    verify(screen).hideInvalidNumberError()
+    verify(screen).hideAnyError()
+  }
+
+  @Test
+  fun `when proceed is clicked with a valid phone number then a network call should be made to check if the phone number is an existing user`() {
+    val inputNumber = "1234567890"
+    whenever(userSession.findExistingUser(inputNumber)).thenReturn(Single.never())
+    whenever(numberValidator.isValid(inputNumber)).thenReturn(true)
+
+    uiEvents.onNext(RegistrationPhoneNumberTextChanged(inputNumber))
+    uiEvents.onNext(RegistrationPhoneDoneClicked())
+
+    verify(screen).showProgressIndicator()
+    verify(userSession).findExistingUser(inputNumber)
+  }
+
+  @Test
+  fun `when the network call for checking phone number fails then an error should be shown`() {
+    val inputNumber = "1234567890"
+
+    whenever(userSession.findExistingUser(inputNumber))
+        .thenReturn(Single.just(FindUserResult.UnexpectedError()))
+        .thenReturn(Single.just(FindUserResult.NetworkError()))
+
+    whenever(numberValidator.isValid(inputNumber)).thenReturn(true)
+
+    uiEvents.onNext(RegistrationPhoneNumberTextChanged(inputNumber))
+    uiEvents.onNext(RegistrationPhoneDoneClicked())
+    uiEvents.onNext(RegistrationPhoneDoneClicked())
+
+    verify(screen, times(2)).showProgressIndicator()
+    verify(screen, times(2)).hideProgressIndicator()
+    verify(screen).showUnexpectedErrorMessage()
+    verify(screen).showNetworkErrorMessage()
+  }
+
+  @Test
+  fun `when the phone number is an existing user then an ongoing login entry should be created and login PIN entry screen should be opened`() {
```

Comment:
Naming: Can we change the test name to ``` `when the phone number belongs to an existing user...` ```?

Reply:
THANK YOU

Reply:
I couldn't think of better words at the time of writing this

Reply:
Done